provisioning/hare_setup.py

```diff
@@ -0,0 +1,383 @@
+                exit(-1)
+            if checkRpm('cortx-s3server') != 0:
+                logging.warning('\'cortx-s3server\' is not installed')
+        except Exception as error:
+            logging.error('Error while checking installed rpms (%s)',
+                          error)
+            exit(-1)
+
+class Init(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            if is_cluster_running() == True:
+                logging.debug('Cluster is already running')
+                if shutdown_cluster() != 0:
+                    logging.error('Failed to shutdown the cluster')
+                    exit(-1)
+                
+            if bootstrap_cluster() != 0:
+                logging.error('Failed to bootstrap the cluster')
+                exit(-1)
+            if shutdown_cluster() != 0:
+                logging.error('Failed to shutdown the cluster')
+                exit(-1)
+        except Exception as error:
+            logging.error('Error while initializing the cluster (%s)',
+                          error)
+            exit(-1)
+
+class Test(argparse.Action):
+    """
+    Read CDF file and compare fields with output of 'hctl status'
+    Will start the cluster(and shutdown before exiting) if not already started
+    """
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            cluster_started = False
+            cluster_status_failure = False
+            if is_cluster_running() == False:
+                if bootstrap_cluster() != 0:
+                    logging.error('Failed to bootstrap the cluster')
+                    exit(-1)
+                cluster_started = True
+
+            if check_cluster_status() != 0:
+                logging.error('Cluster status reports failure')
+                cluster_status_failure = True
+
+            if cluster_started == True:
+                if shutdown_cluster() != 0:
+                    logging.error('Failed to shutdown the cluster')
+                    exit(-1)
+
+            if cluster_status_failure == True:
+                exit(-1)
+
+        except Exception as error:
+            logging.error('Error while checking cluster status (%s)',
+                          error)
+            exit(-1)
+
+class SupportBundle(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            # Default target directory is /tmp/hare
+            if os.system('hctl reportbug') != 0:
+                logging.error('Failed to generate support bundle')
+                exit(-1)
+        except Exception as error:
+            logging.error('Error while generating support bundle (%s)',
+                          error)
+            exit(-1)
+
+class Cleanup(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            logging.info('No-op for Cleanup/Reset as of now for Hare ')
```

Comment:
I don't think this log message is needed, it does not provide any consumable information as such.

Reply:
@SwapnilGaonkar7, I think you removed `Cleanup` class altogether, we need to support the cleanup function, just that it can be a no-op for now. In the above comment I just asked you to remove the log message as it did not provide much information.

Reply:
Done